internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3792,3 +3858,22 @@ func flattenKubernetesClusterAzureMonitorProfile(input *managedclusters.ManagedC
 		},
 	}
 }
+
+func retrySystemNodePoolCreation(ctx context.Context, client *agentpools.AgentPoolsClient, id agentpools.AgentPoolId, profile agentpools.AgentPool) error {
+	// retries the creation of a system node pool 3 times
+	retries := 3
+	attempt := 0
+	for attempt < retries {
+		if err := client.CreateOrUpdateThenPoll(ctx, id, profile); err == nil {
+			break
+		} else {
+			// only return the error on the final retry
+			if attempt == 2 {
+				return err
+			}
+
+			attempt++
+		}
+	}
+	return nil
```

Comment:
would this be better as?
```suggestion
	var err error
	for attempt := 1; attempt < 3; attempt++ {
		if err = client.CreateOrUpdateThenPoll(ctx, id, profile); err == nil {
			return nil
		}
	}
	
	return err
```